src/ServiceControl/Operations/ImportFailedAudits.cs

```diff
@@ -40,14 +42,16 @@ public Task Run(CancellationTokenSource tokenSource)
                         try
                         {
                             var messageContext = new MessageContext(dto.Id, dto.Headers, dto.Body, EmptyTransaction, EmptyTokenSource, EmptyContextBag);
-                            var entity = await auditImporter.ConvertToSaveMessage(messageContext)
-                                .ConfigureAwait(false);
-                            using (var storeSession = store.OpenAsyncSession())
+
+                            //Due to bug https://github.com/Particular/PlatformDevelopment/issues/2464 some failed error could have ended up in failed audits collection
+                            if (messageContext.Headers.ContainsKey(FaultsHeaderKeys.FailedQ))
```

Comment:
This works under the assumption that only error messages might have been imported into the audit collection and not audit messages into the error collection. Given that assembly scanning with features and the container is pretty deterministic it sounds like a valid assumption to make. We only remove the `FaultsHeaderKeys.FailedQ` when we retry messages so we should be safe?

Reply:
The header is removed in `RetryProcessor::StageMessage` method so it safe to use it here.

Reply:
I added a failing test as the first commit to prove the issue was deterministic. Also, there is `When_a_message_fails_to_import` test in the `Audit` category that proves that the failed audits are stored in the proper collection so we're covered.